tasks/main.yml

```diff
@@ -15,12 +15,10 @@
     dest: '{{conda_env_fq_dest_environment}}'
     mode: 0644
 
-- name: create the conda environment
+- name: '{{conda_env_action_command}} the conda environment'
   become: '{{conda_env_escalate}}'
   become_user: root
-  command: '{{conda_env_bin_dir}}/conda env create -f={{conda_env_fq_dest_environment}} -q -n {{conda_env_name}}'
-  args:
-    creates: '{{conda_env_conda_dir}}/envs/{{conda_env_name}}'
+  command: '{{conda_env_bin_dir}}/conda env {{conda_env_action_command}} -f={{conda_env_fq_dest_environment}} -q -n {{conda_env_name}}'
 
```

Comment:
idempotence?

Reply:
hard to get this right with create/update actions. it was an issue with create as well - if conda fails to install one of the package from environment specification, task would fail with directory created and wont run the task in subsequent runs due to presence of the directory. 

i think it might be better to leave this check on conda, it's not going to create/update the environment/package if its present and matches the environment specification.

thoughts?

Reply:
sold